models/page_views/scratch/snowplow_web_pv_engaged_time.sql

```diff
@@ -17,7 +17,7 @@ select
     -- count(distinct) counts duplicates only once
     -- adding snowplow__min_visit_length accounts for the page view event itself.
 
-  {{ var("snowplow__heartbeat", 10) }} * (count(distinct(floor({{ snowplow_utils.to_unixtstamp('ev.derived_tstamp') }}/{{ var("snowplow__heartbeat", 10) }}))) - 1) + {{ var("snowplow__min_visit_length", 5) }} as engaged_time_in_s
+  {{ var("snowplow__heartbeat", 10) }} * (count(distinct(floor({{ snowplow_utils.to_unixtstamp('ev.dvce_created_tstamp') }}/{{ var("snowplow__heartbeat", 10) }}))) - 1) + {{ var("snowplow__min_visit_length", 5) }} as engaged_time_in_s
```

Comment:
Sent discussions in Slack, but at a high level `dervied_tstamp` is usually a more accurate timestamp for absolute times of events, but relies on the assumption there is little to no delay between the event being sent and the collector adding a timestamp - this is not always the case and I found quite a few examples when working through testing. Because we only care about _relative_ time between events in this case, getting a distinct of the `dvce_timetamps` is actually more likely to be a correct measure.